src/platform/ESP32/nimble/BLEManagerImpl.cpp

```diff
@@ -288,6 +289,12 @@ void BLEManagerImpl::_OnPlatformEvent(const ChipDeviceEvent * event)
         break;
 
     case DeviceEventType::kCHIPoBLEWriteReceived:
+        if (getInstance()->pFlags.Has(getInstance()->PASEFlags::kPASESessionFailed))
```

Comment:
I don't know that this is the right way to do this.  My understanding is that the PASE session is only something that needs to exist during the PASE handshake and that once the handshake is complete, the PASE session is no longer relevant and we should track this as a secure session operating over BTP.

Thus, it seems like the PASE session lifetime does not correctly map to opening and closure of the BTP connection.  Furthermore, `kCHIPoBLEWriteReceived` event also seems incorrect.  Why aren't we closing the connection in the `OnSessionEstablishmentError` delegate or directly from `PASESession::OnMessageReceived`?

Reply:
> I don't know that this is the right way to do this. My understanding is that the PASE session is only something that needs to exist during the PASE handshake and that once the handshake is complete, the PASE session is no longer relevant and we should track this as a secure session operating over BTP.
> 
> Thus, it seems like the PASE session lifetime does not correctly map to opening and closure of the BTP connection. Furthermore, `kCHIPoBLEWriteReceived` event also seems incorrect. Why aren't we closing the connection in the `OnSessionEstablishmentError` delegate or directly from `PASESession::OnMessageReceived`?

PASE session API's and BLE API's are kept isolated from each other. Hence closing connection directly in  `PASESession::OnMessageReceived` is not possible.

Reply:
OK, that's fair.